packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol

```diff
@@ -0,0 +1,369 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Target contract is imported by the `Bridge_Initializer`
+import { Bridge_Initializer } from "test/setup/Bridge_Initializer.sol";
+
+// Target contract dependencies
+import {
+    L2StandardBridgeInterop,
+    InvalidDecimals,
+    InvalidLegacyERC20Address,
+    InvalidSuperchainERC20Address,
+    InvalidTokenPair,
+    IOptimismERC20Factory,
+    MintableAndBurnable
+} from "src/L2/L2StandardBridgeInterop.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { IOptimismMintableERC20 } from "src/universal/IOptimismMintableERC20.sol";
+
+// TODO: Replace Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY with optimismSuperchainERC20Factory
+import { Predeploys } from "src/libraries/Predeploys.sol";
+
+contract L2StandardBridgeInterop_Test is Bridge_Initializer {
+    /// @notice Emitted when a conversion is made.
+    event Converted(address indexed from, address indexed to, address indexed caller, uint256 amount);
+
+    /// @notice Test setup.
+    function setUp() public virtual override {
+        super.enableInterop();
+        super.setUp();
+    }
+
+    /// @notice Helper function to setup a mock and expect a call to it.
+    function _mockAndExpect(address _receiver, bytes memory _calldata, bytes memory _returned) internal {
+        vm.mockCall(_receiver, _calldata, _returned);
+        vm.expectCall(_receiver, _calldata);
+    }
+
+    /// @notice Mock ERC20 decimals
+    function _mockDecimals(address _token, uint8 _decimals) internal {
+        _mockAndExpect(_token, abi.encodeWithSelector(IERC20Metadata.decimals.selector), abi.encode(_decimals));
+    }
+
+    /// @notice Mock ERC165 interface
+    function _mockInterface(address _token, bytes4 _interfaceId, bool _supported) internal {
+        _mockAndExpect(
+            _token, abi.encodeWithSelector(IERC165.supportsInterface.selector, _interfaceId), abi.encode(_supported)
+        );
+    }
+
+    /// @notice Mock factory deployment
+    function _mockDeployments(address _factory, address _token, address _deployed) internal {
+        _mockAndExpect(
+            _factory, abi.encodeWithSelector(IOptimismERC20Factory.deployments.selector, _token), abi.encode(_deployed)
+        );
+    }
+
+    /// @notice Assume a valid address for fuzzing
+    function _assumeAddress(address _address) internal view {
```

Comment:
There is a `forge-std` helper that does this

Reply:
https://github.com/foundry-rs/forge-std/blob/bf6606142994b1e47e2882ce0cd477c020d77623/src/StdCheats.sol#L237